microk8s-resources/actions/common/utils.sh

```diff
@@ -143,13 +143,36 @@ get_ips() {
 
 
 produce_server_cert() {
-    # Produce the server certificate adding the IP passed as a parameter
-    # Parameters:
-    # $1 IP we want in the certificate
+    # Produce the server certificates based on the rendered csr.conf.rendered.
+    # The file csr.conf.rendered is compared with csr.conf to determine if a regeneration of server certs must be done.
+    #
+    # Returns 
+    #  0 if no change
+    #  1 otherwise. 
+
+    render_csr_conf
+    if ! [ -f "${SNAP_DATA}/certs/csr.conf" ];
+    then
+        echo "changeme" >  "${SNAP_DATA}/certs/csr.conf" 
+    fi
+
+    if ! cmp -s "${SNAP_DATA}/certs/csr.conf.rendered" "${SNAP_DATA}/certs/csr.conf"; then
+      cp ${SNAP_DATA}/certs/csr.conf.rendered ${SNAP_DATA}/certs/csr.conf
+      openssl req -new -key ${SNAP_DATA}/certs/server.key -out ${SNAP_DATA}/certs/server.csr -config ${SNAP_DATA}/certs/csr.conf
+      openssl x509 -req -in ${SNAP_DATA}/certs/server.csr -CA ${SNAP_DATA}/certs/ca.crt -CAkey ${SNAP_DATA}/certs/ca.key -CAcreateserial -out ${SNAP_DATA}/certs/server.crt -days 100000 -extensions v3_ext -extfile ${SNAP_DATA}/certs/csr.conf
+      echo "1"
+    else
+      echo "0"
+    fi
+
+}
+
+render_csr_conf() {
+    # Render csr.conf.template to csr.conf.rendered
 
-    local IP_ADDR="$1"
+    local IP_ADDR="$(get_ips)"
```

Comment:
Since we do not get the IPs as arguments the for loop below should be updated:
```
        for ADDR in $(echo "$IP_ADDR"); do
            ips+="${sep}IP.$((i++)) = ${ADDR}"
            sep='\n'
        done
```

Reply:
Aaa i missed this one.  And my workstation only comes with one network so the issue didn't manifest.  I'll work on this one. Thanks!